gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -168,4 +170,32 @@ public static Writable deserializeFromString(Class<? extends Writable> writableC
       closer.close();
     }
   }
+
+  /**
+   * Serializes a given {@link FsPermission} object into a {@link String} and stores it as a key, value pair in the
+   * given {@link State}. This method is meant to be used in conjunction with
+   * {@link #deserializeFsPermission(State, String)}.
+   */
+  public static void serializeFsPermission(State state, String key, FsPermission fsPermissions) {
+    state.setProp(key, String.format("%04o", fsPermissions.toShort()));
+  }
+
+  /**
+   * Serializes a given {@link String} that represents the permissions of a file / directory in octal notation. The
+   * {@link String} is stored in the given {@link State}. This method is meant to be used in conjunction with
+   * {@link #deserializeFsPermission(State, String)}.
+   */
+  public static void serializeOctalPermissions(State state, String key, String octalPermissions) {
```

Comment:
not: can this be renamed to `setOctalPermissions`? It is not actually serializing anything.